src/tablestore.rs

```diff
@@ -371,4 +457,98 @@ mod tests {
         )
         .await;
     }
+
+    async fn create_test_sst(
+        ts: &TableStore,
+        id: SsTableId,
+        data: &[(Vec<u8>, Option<Vec<u8>>)],
+    ) -> SSTableHandle {
+        let mut writer = ts.table_writer(id.clone());
+        for (key, value) in data {
+            writer.add(key, value.as_deref()).await.unwrap();
+        }
+        writer.close().await.unwrap()
+    }
+
+    #[tokio::test]
+    async fn test_cache_with_object_store_deletion() {
```

Comment:
This is a good first test, but since it's reading full SSTs for ranges, I think it passes. I suspect a more complex test that verifies multiple non-contiguous subranges within a single SST range read would fail (see my comment above).